### PR TITLE
[Sync] User Profile Values

### PR DESCRIPTION
**Pull request by nhachicha**
WIP to implement #4645 still waiting for ROS to define the JSON/API

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/GetUserProfileValuesResponse.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        Token refreshToken;
+        String message;
+        try {
+            JSONObject obj = new JSONObject(serverResponse);
+            accessToken = obj.has(JSON_FIELD_ACCESS_TOKEN) ?
+                    Token.from(obj.getJSONObject(JSON_FIELD_ACCESS_TOKEN)) : null;
+            refreshToken = obj.has(JSON_FIELD_REFRESH_TOKEN) ?
+                    Token.from(obj.getJSONObject(JSON_FIELD_REFRESH_TOKEN)) : null;
+            error = null;
+            if (accessToken == null) {
```
Comment (bmeike) [inline realm/realm-library/src/objectServer/java/io/realm/internal/network/GetUserProfileValuesResponse.java]: No biggie, but this seems like what ternery expressions were meant for: ```
message = (accessToken == null) 
    ? "accessToken == null"
    : String.format(...);

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java [hunk 2]
```diff
@@ -108,6 +109,28 @@ public ChangePasswordResponse changePassword(Token adminToken, String userId, St
         }
     }
 
+    @Override
+    public GetUserProfileValuesResponse getUserProfileValues(Token userToken, URL profileValuesUrl) {
+        //TODO revisit once the EndPoint is defined/known
+        try {
+            String requestBody = GetUserProfileValuesRequest.create(userToken).toJson();
```
Comment (bmeike) [inline realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java]: Another "no biggie".  I think it future-proofs code like this to have a `return` statement at the end.  What would you think of pulling the definition of `requestBody` out of the `try` block, assigning it in the try, and returning based on it, outside the `try`, at the end of the method?

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/SetUserProfileValuesRequest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    private SetUserProfileValuesRequest(String token, Map<String,String> values) {
+        this.token = token;
+        this.values = values;
+    }
+
+    /**
+     * Converts the request into a JSON payload.
+     */
+    public String toJson() {
```
Comment (bmeike) [inline realm/realm-library/src/objectServer/java/io/realm/internal/network/SetUserProfileValuesRequest.java]: .... again.  I'm guessing this should be an interface...

**Comment by cmelchior** [comment]
Replaced by `RealmUser` in 10, so closing as outdated.

**Pull request closed without merging**